packages/python/plotly/plotly/express/_core.py

```diff
@@ -2440,27 +2439,31 @@ def get_groups_and_orders(args, grouper):
             else:
                 orders[col] = list(OrderedDict.fromkeys(list(orders[col]) + uniques))
 
-    grp_to_idx = {k: i for i, k in enumerate(orders)}
-
     if len(single_group_name) == len(grouper):
         # we have a single group, so we can skip all group-by operations!
         groups = {tuple(single_group_name): df}
     else:
-        required_grouper = list(orders.keys())
+        required_grouper = [
+            key for key in orders if key in grouper and key != one_group
```

Comment:
I believe `one_group` will never be in `orders` so this can be simplified to 
```suggestion
            key for key in orders if key in grouper
```